nil/getting-started/essentials/creating-a-wallet.mdx

```diff
@@ -0,0 +1,64 @@
+# Creating a wallet
+
+This tutorial defines how wallets work in =nil; and outlines the different ways for creating a new wallet.
+
+## Definition
+
+In =nil; a wallet is any smart contract that is able to pay for interactions with real-life users.
+
+The traditional definition of a wallet in Ethereum mostly refers to externally-owned accounts (EOAs) that are secured via private keys and natively hold information about user's balances.
+
+Wallets in =nil; differ from this definition. A =nil; wallet is always a smart contract, and it must be deployed as such. The wallet handles all the usual functionalities (e.g., authenticating requests) but it can also include complex logic such as multi-sig operations.
+
+This means that in almost all cases an external message has to go through a wallet that signs the message and assigns a value to it. The message is only forwarded to its target smart contract for execution only after this process is complete.
+
+:::info[Counterfactual contracts]
+
+It is possible to send funds to an address where a contract is supposed to be deployed. These funds will be stored at that address until the contract is finally deployed, at which point the contract will be able to access them.
+
+Funds sent to an address without a deployed contract are safe, and another party cannot claim them by deploying another contract before the owners of the funds is able to deploy their own contract.
+
+:::
+
+## New wallet creation
+
+### Via the =nil; CLI
+
+The =nil; CLI offers a quick and convenient way to create a new wallet:
+
+```bash
+nil_cli new-wallet
+```
+
+This will produce the following response:
+
+```bash
+New wallet address: NEW_WALLET_ADDRESS
+```
+
+This wallet essentially acts as a typical EOA: it supports basic asset management and authentication operations. The wallet should also be initialized with an initial balance. Send the following request to check this:
+
+```bash
+curl -X POST NIL_ENDPOINT -H "Content-Type: application/json" -d '{
```

Comment:
I believe there are now:

```
nil_cli wallet new
nil_cli wallet balance
```

So there's no need to show this HTTP request, it can be done via CLI